itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenarios.java

```diff
@@ -589,6 +590,72 @@ public void testIncrementalAdds() throws IOException {
   }
 
   @Test
+  public void testIncrementalLoadWithVariableLengthEventId() throws IOException, TException {
+    String testName = "incrementalLoadWithVariableLengthEventId";
+    String dbName = createDB(testName);
+    run("CREATE TABLE " + dbName + ".unptned(a string) STORED AS TEXTFILE");
+
+    String[] empty = new String[] {};
+    advanceDumpDir();
+    run("REPL DUMP " + dbName);
+    String replDumpLocn = getResult(0, 0);
+    String replDumpId = getResult(0, 1, true);
+    LOG.info("Bootstrap-Dump: Dumped to {} with id {}", replDumpLocn, replDumpId);
+    run("REPL LOAD " + dbName + "_dupe FROM '" + replDumpLocn + "'");
+
+    // Each insert generates 3 events. So, in total 15 events gets generated by this loop.
+    for (int i = 0; i < 5; i++) {
+      run("INSERT INTO TABLE " + dbName + ".unptned values('" + Integer.toString(i) + "')");
+    }
+
+    verifySetup("SELECT count(a) from " + dbName + ".unptned", new String[] {"5"});
+
+    // Truncate at the end so that the data set is empty after incremental load.
+    run("TRUNCATE TABLE " + dbName + ".unptned");
+    verifySetup("SELECT a from " + dbName + ".unptned", empty);
+
+    // Inject a behaviour where the event ID starts with 90 and hence overflow across 100 boundary.
+    // This enesures variable length of event ID in the incremental dump
+    BehaviourInjection<NotificationEventResponse,NotificationEventResponse> eventIdModifier
+            = new BehaviourInjection<NotificationEventResponse,NotificationEventResponse>(){
+      private int eventIdIter = 90;
+      @Nullable
+      @Override
+      public NotificationEventResponse apply(@Nullable NotificationEventResponse eventIdList) {
+        if (null != eventIdList) {
+          List<NotificationEvent> eventIds = eventIdList.getEvents();
+          for (int i = 0; i < eventIds.size(); i++) {
+            eventIds.get(i).setEventId(eventIdIter++);
+          }
+          injectionPathCalled = true;
+          return new NotificationEventResponse(eventIds);
+        } else {
+          return eventIdList;
+        }
+      }
+    };
+    InjectableBehaviourObjectStore.setGetNextNotificationBehaviour(eventIdModifier);
+
+    advanceDumpDir();
+    String cmd = "REPL DUMP " + dbName + " FROM " + replDumpId;
+    if (metaStoreClient.getCurrentNotificationEventId().getEventId() < 150) {
+      cmd = cmd + " TO 150";
```

Comment:
If we don't specify the TO clause, then internally we set it to currentNotificationEventId from metastore. If the currentNotificationEventId is less than 100, then test will fail. This can happen randomly. So, just having this if clause to be safe.